### PR TITLE
3531 Add args to subclass of CacheDataset

### LINKED ISSUE #3531: Missing `copy_cache` and `progress` args in several subclass of CacheDataset
**Is your feature request related to a problem? Please describe.**
We added `copy_cache` and `progress` args in `CacheDataset` for better performance and user experience:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/dataset.py#L673.
Would be nice to also add them in the subclasses of `CacheDataset`.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #3531 .

### Description
This PR added new args of `CacheDataset` to subclasses.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/apps/datasets.py [hunk 1]
```diff
@@ -51,6 +51,12 @@ class MedNISTDataset(Randomizable, CacheDataset):
             will take the minimum of (cache_num, data_length x cache_rate, data_length).
         num_workers: the number of worker threads to use.
             if 0 a single thread will be used. Default is 0.
+        progress: whether to display a progress bar.
```
Comment (wyli) [inline monai/apps/datasets.py:54]: Would be misleading for the progress bar for downloading, could you review this?

Comment (Nic-Ma) [reply monai/apps/datasets.py]: That's a good point! enhanced the description in the latest commit.

Thanks.

**Review by wyli — APPROVED** [review]
Looks good, note that the download utility also accepts a progress bar flag

**Comment by Nic-Ma** [comment]
> Looks good, note that the download utility also accepts a progress bar flag

Hi @wyli ,

Do you think it's better to apply this progress bar arg for both CacheDataset and download?

Thanks.

**Comment by wyli** [comment]
> > Looks good, note that the download utility also accepts a progress bar flag
> 
> 
> 
> Hi @wyli ,
> 
> 
> 
> Do you think it's better to apply this progress bar arg for both CacheDataset and download?
> 
> 
> 
> Thanks.
> 
> 

Im fine with either option, as long as the docstring is consistent with it...

**Comment by Nic-Ma** [comment]
Hi @wyli ,

I updated the logic to apply `progress` arg for both.

Thanks.

**Comment by Nic-Ma** [comment]
Mark the Blossom error for future reference:
```
2021-12-22T14:40:04.5597947Z Traceback (most recent call last):
2021-12-22T14:40:04.5599020Z   File "/opt/conda/lib/python3.8/site-packages/parameterized/parameterized.py", line 533, in standalone_func
2021-12-22T14:40:04.5599815Z     return func(*(a + p.args), **p.kwargs)
2021-12-22T14:40:04.5600883Z   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/tests/test_cucim_transform.py", line 79, in test_tramsforms_numpy_single
2021-12-22T14:40:04.5601701Z     output = CuCIM(**params)(input)
2021-12-22T14:40:04.5602662Z   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/transforms/utility/array.py", line 1216, in __call__
2021-12-22T14:40:04.5603525Z     return self.transform(data, *self.args, **self.kwargs)
2021-12-22T14:40:04.5604556Z   File "/opt/conda/lib/python3.8/site-packages/cucim/core/operations/intensity/zoom.py", line 153, in zoom
2021-12-22T14:40:04.5605353Z     kernel = CUDA_KERNELS.get_function("zoom_out_kernel")
2021-12-22T14:40:04.5606092Z   File "cupy/_core/raw.pyx", line 485, in cupy._core.raw.RawModule.get_function
2021-12-22T14:40:04.5606984Z   File "cupy/_core/raw.pyx", line 96, in cupy._core.raw.RawKernel.kernel.__get__
2021-12-22T14:40:04.5607845Z   File "cupy/_core/raw.pyx", line 113, in cupy._core.raw.RawKernel._kernel
2021-12-22T14:40:04.5608663Z   File "cupy/_util.pyx", line 59, in cupy._util.memoize.decorator.ret
2021-12-22T14:40:04.5609428Z   File "cupy/_core/raw.pyx", line 547, in cupy._core.raw._get_raw_module
2021-12-22T14:40:04.5610153Z   File "cupy/_core/core.pyx", line 1961, in cupy._core.core.compile_with_cache
2021-12-22T14:40:04.5610923Z   File "cupy/_core/core.pyx", line 2030, in cupy._core.core.compile_with_cache
2021-12-22T14:40:04.5612189Z   File "/opt/conda/lib/python3.8/site-packages/cupy/cuda/compiler.py", line 434, in compile_with_cache
2021-12-22T14:40:04.5612955Z     return _compile_with_cache_cuda(
2021-12-22T14:40:04.5613953Z   File "/opt/conda/lib/python3.8/site-packages/cupy/cuda/compiler.py", line 502, in _compile_with_cache_cuda
2021-12-22T14:40:04.5614666Z     mod.load(cubin)
2021-12-22T14:40:04.5615438Z   File "cupy/cuda/function.pyx", line 241, in cupy.cuda.function.Module.load
2021-12-22T14:40:04.5616490Z   File "cupy/cuda/function.pyx", line 243, in cupy.cuda.function.Module.load
2021-12-22T14:40:04.5617757Z   File "cupy_backends/cuda/api/driver.pyx", line 253, in cupy_backends.cuda.api.driver.moduleLoadData
2021-12-22T14:40:04.5619107Z   File "cupy_backends/cuda/api/driver.pyx", line 125, in cupy_backends.cuda.api.driver.check_status
2021-12-22T14:40:04.5620761Z cupy_backends.cuda.api.driver.CUDADriverError: CUDA_ERROR_UNSUPPORTED_PTX_VERSION: the provided PTX was compiled with an unsupported toolchain.
```

Thanks.

**Pull request merged**